### PR TITLE
ISPN-12039 Remote Per-Cache Simple Tutorial

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-12039

### CODE REVIEW

File: remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build());
+      // Obtain a remote cache that does not exist.
+      // Rather than return null, create the cache from a template.
+      RemoteCache<String, String> cache = cacheManager.getCache("my-cache");
+      /// Store a value
+      cache.put("hello", "world");
+      // Retrieve the value and print it out
+      System.out.printf("key = %s\n", cache.get("key"));
+      // Stop the cache manager and release all resources
+      cacheManager.stop();
```
Comment (pruivo) [inline remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:52]: you can use try-with-resources to automatically close the client when finishes.
```java
try (RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build())) {
    ...
}
```

File: remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+               .configuration("<infinispan><cache-container><distributed-cache name=\"another-cache\"><encoding media-type=\"application/x-protostream\"/></distributed-cache></cache-container></infinispan>");
+      // Connect to the server
+      RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build());
+      // Obtain a remote cache that does not exist.
+      // Rather than return null, create the cache from a template.
+      RemoteCache<String, String> cache = cacheManager.getCache("my-cache");
+      /// Store a value
+      cache.put("hello", "world");
+      // Retrieve the value and print it out
+      System.out.printf("key = %s\n", cache.get("key"));
```
Comment (pruivo) [inline remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:50]:
```suggestion
      System.out.printf("key = %s\n", cache.get("hello"));
```

File: remote-per-cache/pom.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <executable>java</executable>
+                    <arguments>
+                        <argument>-Djava.net.preferIPv4Stack=true</argument>
+                        <argument>-Djava.util.logging.config.file=src/main/resources/logging.properties</argument>
```
Comment (pruivo) [inline remote-per-cache/pom.xml:29]: the file doesn't exist. the argument can be removed.

File: remote-per-cache/pom.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    <execution>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <executable>java</executable>
+                    <arguments>
+                        <argument>-Djava.net.preferIPv4Stack=true</argument>
```
Comment (pruivo) [inline remote-per-cache/pom.xml:28]: it only affects JGroups. can be removed.

Comment (oraNod) [reply remote-per-cache/pom.xml:28]: removed. thanks!

File: remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java (hunk 1 above)
Comment (oraNod) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java]: updated with try-with-resources. seems to work nicely. thanks!

File: remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+               .saslMechanism("DIGEST-MD5");
+      //Add per-cache configuration that uses an org.infinispan cache template.
+      builder.remoteCache("my-cache")
+               .templateName("org.infinispan.DIST_SYNC");
+      //Add per-cache configuration with a cache definition in XML format.
+      builder.remoteCache("another-cache")
+               .configuration("<infinispan><cache-container><distributed-cache name=\"another-cache\"><encoding media-type=\"application/x-protostream\"/></distributed-cache></cache-container></infinispan>");
+      // Connect to the server
+      try (RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build())) {
+      // Obtain a remote cache that does not exist.
```
Comment (pruivo) [inline remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java:44]: nitpick: indentation.

Comment (oraNod) [reply remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java]: that's a good nitpick though. should be sorted now.

**Comment by pruivo** [comment]
integrated! thanks @oraNod !

**Pull request merged**